pkg/cluster/kubernetes/talos_managed.go

```diff
@@ -288,3 +350,112 @@ func checkPodStatus(ctx context.Context, cluster UpgradeProvider, service, node,
 
 	return nil
 }
+
+//nolint:gocyclo
+func checkDeprecated(ctx context.Context, cluster UpgradeProvider, options UpgradeOptions) error {
+	// disable kubernetes deprecations warnings
+	klog.LogToStderr(false)
+
+	options.Log("checking for resource APIs to be deprecated in version %s", options.ToVersion)
+
+	config, err := cluster.K8sRestConfig(ctx)
+	if err != nil {
+		return err
+	}
+
+	k8sClient, err := dynamic.NewForConfig(config)
+	if err != nil {
+		return fmt.Errorf("error building kubernetes client: %w", err)
+	}
+
+	staticClient, err := cluster.K8sHelper(ctx)
+	if err != nil {
+		return fmt.Errorf("error building kubernetes client: %s", err)
+	}
+
+	hasDeprecated := false
+
+	warnings := bytes.NewBuffer([]byte{})
+
+	w := tabwriter.NewWriter(warnings, 0, 0, 3, ' ', 0)
+
+	resources, ok := deprecations[options.Path()]
+	if ok {
+		for _, resource := range resources {
+			gvr, _ := schema.ParseResourceArg(resource)
+
+			if gvr == nil {
+				return fmt.Errorf("failed to parse group version resource %s", resource)
+			}
+
+			var (
+				res        *unstructured.UnstructuredList
+				namespaces *v1.NamespaceList
+			)
+
+			count := 0
+
+			namespaces, err = staticClient.CoreV1().Namespaces().List(ctx, metav1.ListOptions{})
+			if err != nil {
+				return err
+			}
+
+			probeResources := func(namespace *v1.Namespace) error {
```

Comment:
```suggestion
			probeResources := func(namespace v1.Namespace) error {
```

this would allow to remove `ns := ns` below